src/components/Input/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,135 @@
+    /** The input error field context (critical) */
+    context: ErrorContext;
+    /** This message will be rendered under the input when context critical will be applied */
+    errorMessage: string;
+};
+
+interface BaseProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size'> {
+    /** Should the input field be disabled or not */
+    disabled?: boolean;
+    /** Should the input field be readOnly or not */
+    readOnly?: boolean;
+    /** Whether or not to show block-level input field (full width) */
+    isBlock?: boolean;
+    /** The size of the input field */
+    size?: InputSize;
+    /** Field label */
+    label?: string;
+    /** Helper text under the input, display the error when the context is critical */
+    helperText?: string;
+    /**  Controls whether space is reserved for error messages under the input field when validation is expected
+     *  to avoid "jumping" UI. */
+    reserveErrorMessageSpace?: boolean;
+}
+
+export type Props = BaseProps & (ErrorStateProps | { context?: undefined; errorMessage?: never });
+
+const { block, elem } = bem('SearchInput', styles);
+
+export const SearchInput = React.forwardRef<HTMLInputElement, Props>(
+    (
+        {
+            id,
+            children,
+            context = undefined,
+            disabled = false,
+            readOnly = false,
+            isBlock = false,
+            size = 'medium',
+            value: initialValue = '',
+            label,
+            helperText,
+            errorMessage,
+            reserveErrorMessageSpace = false,
+            ...rest
+        },
+        ref
+    ) => {
+        const fallbackId = React.useId();
+        const idRef = id || fallbackId;
+        const [value, setValue] = React.useState(initialValue);
+        const [hasText, setHasText] = React.useState<boolean>(!!initialValue);
+
+        const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+            setValue(e.target.value);
+            setHasText(!!e.target.value);
+        };
+
+        // Function to clear the input value
+        const handleClear = () => {
+            setValue('');
+            setHasText(false);
+        };
+
+        return (
+            <>
+                {label && (
+                    <label {...elem('label')} htmlFor={idRef}>
+                        {label}
+                    </label>
+                )}
+                <div
+                    data-testid="searchInputContainer"
+                    {...block({ withIcon: true, context, size, isBlock, disabled, readOnly })}
+                >
+                    <Search
+                        {...elem('icon', { type: 'search', bold: hasText, size })}
+                        viewBox="0 0 24 24"
+                    />
+                    <input
+                        {...elem('input', { size })}
+                        {...rest}
+                        id={idRef}
+                        ref={ref}
+                        type="text"
```

Comment:
why using type `text` for a `search` input?

Reply:
type="search" adds a 'clear' icon itself. That's why I chose to go with text instead. I could add something like this:
input[type="search"]::-webkit-search-cancel-button {
  -webkit-appearance: none;
  height: 1em;
  width: 1em;
  border-radius: 50em;
  background: url(https://pro.fontawesome.com/releases/v5.10.0/svgs/solid/times-circle.svg) no-repeat 50% 50%;
  background-size: contain;
  opacity: 0;
  pointer-events: none;
}

To get rid of the button... But i just didn't think that was the neatest way of doing things.

Reply:
I think you can use `appearance: none` (see also Checkbox and Radio button uses that) to cancel browser specific looks and add your own. 
I think `type="search"` does more then the clear button though. E.g. on mobile (which we might want to support one day) it changes the enter button; also there is a different behaviour about caching things people typed (that's for desktop too).